### ISSUE TITLE
PCMPEQB, PCMPEQW and PCMPEQD

### ISSUE THREAD

**Issue by JonathanSalwan** [JonathanSalwan/Triton#221]
related to #210

**Comment by 0ca** [comment]
I tried the test script `test/test_semantics.py` in 32 bits with the next code:
```
#include <stdio.h>
#include <string.h>

int main(int argc, const char **argv)
{
    char *arg1 = argv[1];
    if (strlen(arg1) == 2)
        printf("Correct len\n");
    return 0;
}
```

And the instruction pcmpeqb isn't symbolic executed correctly. The symbolic value is 128 bits but the concrete value is 64 bits long.
```
[KO] 0xb3f6a9ca: pcmpeqb xmm0, xmm1 (1 error)
     Register       : xmm0
     Symbolic Value : ffffffffffffffffffffffff0000ff00
     Concrete Value : 000000000000ff00
     Expression     : (concat (ite (= ((_ extract 127 120) ((_ extract 127 0) (_ bv0 128))) ((_ extract 127 120) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 119 112) ((_ extract 127 0) (_ bv0 128))) ((_ extract 119 112) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 111 104) ((_ extract 127 0) (_ bv0 128))) ((_ extract 111 104) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 103 96) ((_ extract 127 0) (_ bv0 128))) ((_ extract 103 96) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 95 88) ((_ extract 127 0) (_ bv0 128))) ((_ extract 95 88) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 87 80) ((_ extract 127 0) (_ bv0 128))) ((_ extract 87 80) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 79 72) ((_ extract 127 0) (_ bv0 128))) ((_ extract 79 72) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 71 64) ((_ extract 127 0) (_ bv0 128))) ((_ extract 71 64) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 63 56) ((_ extract 127 0) (_ bv0 128))) ((_ extract 63 56) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 55 48) ((_ extract 127 0) (_ bv0 128))) ((_ extract 55 48) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 47 40) ((_ extract 127 0) (_ bv0 128))) ((_ extract 47 40) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 39 32) ((_ extract 127 0) (_ bv0 128))) ((_ extract 39 32) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 31 24) ((_ extract 127 0) (_ bv0 128))) ((_ extract 31 24) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 23 16) ((_ extract 127 0) (_ bv0 128))) ((_ extract 23 16) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 15 8) ((_ extract 127 0) (_ bv0 128))) ((_ extract 15 8) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)) (ite (= ((_ extract 7 0) ((_ extract 127 0) (_ bv0 128))) ((_ extract 7 0) ((_ extract 127 0) (_ bv1229979745 128)))) (_ bv255 8) (_ bv0 8)))
```
The instruction `movdqu` has the same problem:
```
[KO] 0xb3f6a9c6: movdqu xmm1, xmmword ptr [edi] (1 error)
     Register       : xmm1
     Symbolic Value : 5242494c5f444c5f4d565f4e49500061
     Concrete Value : 0000000049500061
     Expression     : (concat ((_ extract 7 0) (_ bv82 8)) ((_ extract 7 0) (_ bv66 8)) ((_ extract 7 0) (_ bv73 8)) ((_ extract 7 0) (_ bv76 8)) ((_ extract 7 0) (_ bv95 8)) ((_ extract 7 0) (_ bv68 8)) ((_ extract 7 0) (_ bv76 8)) ((_ extract 7 0) (_ bv95 8)) ((_ extract 7 0) (_ bv77 8)) ((_ extract 7 0) (_ bv86 8)) ((_ extract 7 0) (_ bv95 8)) ((_ extract 7 0) (_ bv78 8)) ((_ extract 7 0) (_ bv73 8)) ((_ extract 7 0) (_ bv80 8)) ((_ extract 7 0) (_ bv0 8)) ((_ extract 7 0) (_ bv97 8)))
```

**Comment by JonathanSalwan** [comment]
Fixed in the v0.3
